airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -187,8 +188,9 @@ export const AuthenticationProvider: React.FC<React.PropsWithChildren<unknown>>
       hasPasswordLogin(): boolean {
         return !!state.providers?.includes("password");
       },
-      hasCorporateEmail(): boolean {
-        return !FREE_EMAIL_SERVICE_PROVIDERS.some((provider) => state.currentUser?.email.endsWith(`@${provider}`));
+      hasCorporateEmail(email?: string): boolean {
+        const userEmail = email ?? state.currentUser?.email;
+        return !FREE_EMAIL_SERVICE_PROVIDERS.some((provider) => userEmail?.endsWith(`@${provider}`));
```

Comment:
Code style suggestion: adding this as the default value for the parameter saves us another variable for it.

```suggestion
      hasCorporateEmail(email: string | undefined = state.currentUser?.email): boolean {
        return !FREE_EMAIL_SERVICE_PROVIDERS.some((provider) => email?.endsWith(`@${provider}`));
```